### ISSUE TITLE
Table width more than window size

### ISSUE THREAD

**Issue by jamieabc** [zippyui/react-datagrid#56]
If there are too many columns in a table, is there any way to setup datable total width and customize horizontal scrollbar?

**Comment by radubrehar** [comment]
You can specify style.width on the DataGrid. Also you can specify column.width or column.flex to adjust different column sizes.

**Comment by jamieabc** [comment]
I just tried style.width on DataGrid but seems not working, did I use it right?

    <DataGrid
      ...
      style={{width: 5000}}
      ...
    />

**Comment by radubrehar** [comment]
The horiz scroll on mac for now needs shift+scroll. See #43

**Comment by jamieabc** [comment]
Does it working now? I use Chrome browser and press `shift` with trackpad & mouse scroll not working?

**Comment by zippyui-react** [comment]
Yesd, you can test it on http://zippyui.com/react-datagrid/#/examples/column-resizing

**Comment by jamieabc** [comment]
Does it make any difference if I use  `<DataGrid style={{width: 3000}}  ... />`? 

Currently it looks like  horizontal scrolling working when width setting in `column` but not in `DataGrid`?

**Comment by zippyui-react** [comment]
By default columns are flexed. So if you have 5 columns in the grid, they will split available width among them, so you'll never have a horiz scrollbar. If instead you set their widths explicitly, if they expand beyond available width, the horiz scrollbar will show.

**Comment by jamieabc** [comment]
If I set `<DataGrid style={{width: 2000}} .../>` and set `onColumnResize` callback with follows:

    this.onColumnResize = (firstCol, firstSize, secondCol, secondSize) =>  {
      let windowSize = $(window).width();
      let newTableWidth = this.refs.DataGrid.props.style.width + (firstSize - firstCol.width);
      let oldTableWidth = this.refs.DataGrid.props.style.width;
      this.refs.DataGrid.props.style.width = newTableWidth < windowSize ? oldTableWidth : newTableWidth;
      console.log("table old width " + oldTableWidth + ', new width ' + newTableWidth);
      firstCol.width = firstSize;
      this.setState({});
    }

Basically increase total table width but not decrease table width less than current window size. But if I increase table with over original table width, e.g. 2000, right most column will be truncated (some words not seeing).

Is there any way that table render for total width larger than origin width?

**Comment by radubrehar** [comment]
`this.refs.DataGrid.props.style.width` will not no anything. This is not how React works.

You should re-render the app, with a controlled value for style.width.

**Comment by jamieabc** [comment]
Is it possible that I want to have a horizontal scrollbar and set `<DataGrid style={{width: 1000}} />`? Which means that scrollbar and increase/decrease total table width coexist?

**Comment by radubrehar** [comment]
The horiz scrollbar is dictated by the dimensions of the columns.  If you have 3 cols with width 500, and have the grid style width to 1000, the scrollbar will apear.

**Comment by jamieabc** [comment]
It's quite a large repository and not put on github, but I upload a pic of scrollbar cover checkbox at [here](https://github.com/jamieabc/webpack-example/blob/datagrid/pic.png). From dev tool, the scrollbar is cover the last row of checkbox, so currently cannot check

**Comment by jamieabc** [comment]
Is is possible that path new total table with into `Datagrid setState`? Like following:

    class DataTable extends React.Component {
      constructor () {
        this.onColumnResize = (firstCol, firstSize, secondCol, secondSize) => {
          firCol.width = firstSize;
          let newWidth = calculateNewWidth = (...) // this function calculate new table width by adding each column width
          this.setState({   ???? = newWidth });
        }
      }
    }

If there's any key that maps this setting?

**Comment by radubrehar** [comment]
There is a 20px space reserved for the vertical scrollbar.

**Comment by jamieabc** [comment]
Is there any way to let horizontal scroll bar always exist in Chrome?

**Comment by radubrehar** [comment]
The horiz scrollbar will scroll on shift+scroll, but does not need shift to show.

**Comment by jamieabc** [comment]
Yeah, I am wondering that if user didn't aware that horizontal scroll need to press `shift` on mac, at least show horizontal scrollbar to let user to drag. 

Because like me didn't notice `shift` at first, don't know how to do horizontal scroll.

I can build local version of react-datagrid, but don't know where to modify this behavior.

**Comment by radubrehar** [comment]
We have an issue on that, #43 and will try and fix it soon.

**Comment by jamieabc** [comment]
Oh~~that's great!! Many appreciate

**Comment by jamieabc** [comment]
By the way, I just find how to let Mac Chrome always show horizontal scroll bar, it's a system setting in OS X, please refer to [here](http://kb.tableau.com/articles/knowledgebase/displaying-scroll-bar)

**Comment by radubrehar** [comment]
We don't want our users to need to modify a setting in their OS. If that helps you, that's fine :)

**Comment by jamieabc** [comment]
Oh, yeah, it would be really great if data grid can handle trackpad horizontal event. And I can temporarily turn on this setting for now. Thanks.

**Comment by jamieabc** [comment]
I also find out another solution that doesn't need Mac user modify system settings. I add following custom css style as follows:

    ::-webkit-scrollbar {
        height: 10px;
        background-color: #f5f5f5;
    }
    ::-webkit-scrollbar-thumb {
        border-radius: 1px;
        background-color: rgba(0, 0, 0, .5);
        -webkit-box-shadow: inset 0 0 1px rgba(255, 255, 255, 0.2);
    }
    ::-webkit-scrollbar-track {
        -webkit-box-shadow: inset 0 01px rgba(0, 0, 0, 0,5);
        bordre-radius: 1px;
    }

This makes webkit (chrome) to always show scroll bar if content over current layout size
